### PR TITLE
Add Sphinx documentation for statichit plugin.

### PR DESCRIPTION
**Pull request by ywkaras**
Also remove obsolete readme file for statichit.  Also improve Au test for statichit.

### CODE REVIEW

File: doc/admin-guide/plugins/statichit.en.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+
+This is a simple plugin to serve static content from the proxy's local filesystem. It shares some
+of the same functionality as the `healthchecks` plugin, but is a remap plugin (thereby making it
+reloadable). It does not use fsnotify for watching the source files.
```
Comment (mlibbey) [inline doc/admin-guide/plugins/statichit.en.rst:26]: Is this last sentence meant to imply that source file changes will need a reload to begin serving? If so, perhaps say that explicitly? like
"It does not use fsnotify for watching the source files. New content will be served after |TS| reloads."

File: doc/admin-guide/plugins/statichit.en.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+example.  Otherwise, the value must be specified as the next pparam.  `@pparam=-a @pparam=5` for example.
+
+When exact match is enabled, if the request URL has a path, a response with NOT FOUND status will be sent.
+
+Mime Headers in Responses with OK/200 Status
+============================================
+
+#.  Content-Type.  The value given with the --mime-type parameter, or its default value.
+#.  Content-Length.
+#.  Cache-Control.  If --max-age is 0, the value will be "no-cache".  Otherwise, the value will be "max-age=d", where
```
Comment (mlibbey) [inline doc/admin-guide/plugins/statichit.en.rst:66]: Is it really no-cache or no-store? Probably doesn't end up mattering for this use case, as the origin is still ATS, but no-cache is widely misunderstood, and would be good to not add to confusion :)

File: doc/admin-guide/plugins/statichit.en.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+======================== ====================================================
+Name                     Description
+======================== ====================================================
+statichit.response_bytes The total number of bytes emitted
+statichit.response_count The number of HTTP responses generated by the plugin
+======================== ====================================================
+
+remap.config Examples:
+======================
+
```
Comment (mlibbey) [inline doc/admin-guide/plugins/statichit.en.rst:84]: Serve a request for /internal/healthcheck with the contents of the file located at /var/run/trafficserver/healthcheck.txt. If the file is present, send with a Mime-type of text/plain. Als set the Cache-Control: header to no-cache.

File: doc/admin-guide/plugins/statichit.en.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+| map /internal/healthcheck \\
+| http://127.0.0.1 \\
+| @plugin=statichit.so @pparam=--file-path=/var/run/trafficserver/healthcheck.txt \\
+| @pparam=--mime-type=text/plain \\
+| @pparam=--success-code=200 \\
+| @pparam=--failure-code=403 \\
+| @pparam=--max-age=0 \\
+| @pparam=--disable-exact
+|
```
Comment (mlibbey) [inline doc/admin-guide/plugins/statichit.en.rst:93]: Serve a request for content.example.com/content_xyz.txt with the contents of the file located at /opt/ats/etc/trafficserver/content_bodies/content_xyz.txt (as /opt/ats/etc/trafficserver/content_bodies is meant to imply a directory in this example).

File: doc/admin-guide/plugins/statichit.en.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+| map http://content.example.com/content_xyz.txt \\
+| http://127.0.0.1 \\
+| @plugin=statichit.so @pparam=--file-path=/opt/ats/etc/trafficserver/content_bodies \\
+| @pparam=--failure-code=404 \\
+| @pparam=--max-age=604800
+|
```
Comment (mlibbey) [inline doc/admin-guide/plugins/statichit.en.rst:99]: Serve a request for content.example.com/content_abc.json with the contents of the file located at /opt/ats/etc/trafficserver/content_bodies/content_abc.json (as /opt/ats/etc/trafficserver/content_bodies is meant to imply a directory in this example) with a mime-type of application/json. Set cache-control to have a max-age of 7 days (604800 seconds).

**Comment by ywkaras** [comment]
[approve ci docs]

File: doc/admin-guide/plugins/statichit.en.rst (hunk 2 above)
Comment (ywkaras) [reply doc/admin-guide/plugins/statichit.en.rst]: https://github.com/apache/trafficserver/blob/b46f5d061af455dc75956346da3308788381b30d/plugins/statichit/statichit.cc#L337 .  Core TS seems to add a no-store cache control header for 4xx responses.  (May be affected by negative caching config.)

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
This is purely a documentation and testing enhancement.  Backports are desirable but not crucial.

**Comment by JosiahWI** [comment]
[approve ci rocky autest]

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Review by moonchen — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Comment by ezelkow1** [comment]
Do we still need this for 9.2?

**Pull request merged**